### PR TITLE
Better cuckoo report

### PR DESCRIPTION
**Pull request by michaelweiser**
Here are some improvements to the Cuckoo report to showcase what would be involved to implement input validation as suggested in #188. Lightly tested: The testsuite runs. See individual commits for rationale.

### CODE REVIEW

**Comment by michaelweiser** [comment]
Some more testing with the dummy Cuckoo API shows the expected behaviour and warning message:
```
2021-06-29 10:04:41,456 - peekaboo.toolbox.cuckoo - (CuckooJobTracker) - WARNING - Report returned from Cuckoo contained invalid data: signature descriptions are expected to be strings
```
This also revealed missing type checking for the score and the bigger structures of the report - added.

**Comment by michaelweiser** [comment]
Fallout from not storing the full report any more should now be addressed by adding the origin URL of the report to our dump instead.

**Review by Jack28 — APPROVED** [review]
LGTM

File: peekaboo/sample.py [hunk 1]
```diff
@@ -358,7 +358,7 @@ def dump_processing_info(self):
                                          filename + '_cuckoo_report.json')
             try:
                 with open(cuckoo_report, 'wb+') as cr_json_file:
-                    cr_json = json.dumps(self.__cuckoo_report.raw,
```
[the remark this answers is not in the document]
Comment (michaelweiser) [reply peekaboo/sample.py:361]: Right. Done!

**Comment by michaelweiser** [comment]
@jack28: Merging this will create conflicts in #188 and require a rebase. Good to go anyway?

**Comment by Jack28** [comment]
I'm aware of that. Go ahead

**Pull request merged**